util-core/src/main/scala/com/twitter/util/Batcher.scala

```diff
@@ -0,0 +1,30 @@
+package com.twitter.util
+
+import java.util.logging.{Level, Logger}
+
+import scala.collection.mutable
+
+/** Provides a clean, lightweight interface for controlling a BatchExecutor
```

Comment:
No, I think BatchExecutor as a parameter to the Batcher constructor is still a good idea, but Future.batched should be the only way to get a Batcher.  Let's make the constructor private[util]. :ship: